### PR TITLE
Convert Requirements to .md format

### PR DESCRIPTION
**Pull request by RyanBerge**
Consolidate Navigation requirements into a single subsystem
- This will require refactoring of trello back-references
(If this is approved, I will take the necessary steps)

### CODE REVIEW

File: Top-Level Requirements.md [hunk 1]
```diff
@@ -1,63 +1,86 @@
...
+* 1.2 The executive system **shall** start and manage autonomous operation.  
+* 1.3 The executive system **shall** monitor and delegate system responsibilities and state in between mission subsystems.  
+* 1.4 The executive system **shall** support two classes of commands:  
+    * 1.4.1 The Autonomous Command: Commences the main mission phases and tracks autonomous mission state.  
+    * 1.4.2 Teleop Commands: A lightweight command processing mode which contains user commands to be fed to control systems.  
+* 1.5 The executive system **shall** track mission progress and mission time elapsed.  
+* 1.6 The executive system **shall** be able to preempt all subsystem.  
+* 1.7 The executive system **shall** return the system to a safe state upon failure.  
+### 2.  Mission Control Interface
+* 2.1 The control interface **shall** have commands to:
```
Comment (collinsc) [inline Top-Level Requirements.md]: shall allow the operator to?

File: Top-Level Requirements.md [hunk 2]
```diff
@@ -1,63 +1,86 @@
...
+    * 2.1.1 Start autonomous operation.
+    * 2.1.2 Halt autonomous operation.  
+    * 2.1.3 Manually input directional instructions for driving
+    * 2.1.4 Execute a digging procedure
+### 3.  Communication Subsystem
+* 3.1 The communication subsystem **shall** monitor all top level diagnostic communication, and display them in the command center.
```
Comment (collinsc) [inline Top-Level Requirements.md]: top level diagnostic communications?

File: Top-Level Requirements.md [hunk 3]
```diff
@@ -1,63 +1,86 @@
...
+* 3.2 The communication subsystem **shall** render necessary and sufficient data to the command center to control the mission.
+* 3.3 The communication subsystem could monitor compressed forward and rear video feeds when requested.
+* 3.4 The communication subsystem should monitor bandwidth of communications.
```
Comment (collinsc) [inline Top-Level Requirements.md]: 3.3 should have could bolded, and in 3.4 should should become shall

File: Top-Level Requirements.md [hunk 4]
```diff
@@ -1,63 +1,86 @@
...
+* 3.5 The communication subsystem **shall** adhere to NASA rules and regulations:
+    <https://www.nasa.gov/sites/default/files/atoms/files/2018_rulesrubrics_partii.pdf>
+* 3.6 The subsystem could encrypt all incoming and outgoing transmissions
```
Comment (collinsc) [inline Top-Level Requirements.md]: bold could for consistency

File: Top-Level Requirements.md [hunk 5]
```diff
@@ -1,63 +1,86 @@
...
+* 8.3 The mining subsystem **shall** follow a predefined series of mining movements during the mining procedure.
+* 8.4 The mining procedure **shall** define an ordered set of mining locations (and dumping locations) relative to the robot position
+* 8.5 The mining procedure **shall** further be divided into atomically-operating subprocedures (scoops)
+* 8.6 The mining procedure **shall** always check timing flags before beginning scoops
+* 8.7 The mining procedure **shall** be terminated upon detecting timing flags
+* 8.8 Upon completing the mining procedure, the mining subsystem **shall**:
+    * 8.8.1 Define an “obstacle” representing the digging hole and send it to the navigation subsystem.
+    * 8.8.2 Return the mining arm to the starting position.
+### 9.  Dumping Subsystem
+* 9.1 As a precondition the rover should be facing east, relatively close to the dumping receptacle.
```
Comment (collinsc) [inline Top-Level Requirements.md]: bold for consistency

**Review by collinsc — CHANGES_REQUESTED** [review]
I like the reorganization of navigation I think that makes it more readable and clear. 

I spotted a tiny grammar mistake, and one or two cosmetic things for consistency, but other than that looks great. 

Thanks Ryan.

File: Top-Level Requirements.md (hunk 5 above)
Comment (RyanBerge) [reply Top-Level Requirements.md]: I think we can probably make this a **shall**, rather than a **should**.

**Review by collinsc — APPROVED** [review]

File: Top-Level Requirements.md [hunk 6]
```diff
@@ -1,63 +1,86 @@
...
+* 8.1 The mining subsystem **shall** be provided a time limit and not operate longer than that time limit.
+* 8.2 The mining subsystem **shall** hold a predefined mining procedure.
+* 8.3 The mining subsystem **shall** follow a predefined series of mining movements during the mining procedure.
+* 8.4 The mining procedure **shall** define an ordered set of mining locations (and dumping locations) relative to the robot position
+* 8.5 The mining procedure **shall** further be divided into atomically-operating subprocedures (scoops)
+* 8.6 The mining procedure **shall** always check timing flags before beginning scoops
+* 8.7 The mining procedure **shall** be terminated upon detecting timing flags
+* 8.8 Upon completing the mining procedure, the mining subsystem **shall**:
+    * 8.8.1 Define an “obstacle” representing the digging hole and send it to the navigation subsystem.
+    * 8.8.2 Return the mining arm to the starting position.
```
Comment (AdamuKaapan) [inline Top-Level Requirements.md:79]: 8.9 The mining subsystem **shall** provide a method to move the digging arm out of the bin to a predefined dumping position to enable bin movement without interference

Comment (AdamuKaapan) [reply Top-Level Requirements.md:79]: (this is my proposed addition here, feel free to change it to a way that's better phrased)

File: Top-Level Requirements.md [hunk 7]
```diff
@@ -1,63 +1,86 @@
...
+### 9.  Dumping Subsystem
+* 9.1 As a precondition the rover **shall** be facing east, relatively close to the dumping receptacle.
+* 9.2 The dumping subsystem **shall** perform the localization procedure prior to the start of the  dumping procedure.
+* 9.3 The dumping procedure **shall**:
+    * 9.3.1 Orient itself and turn to face east.
+    * 9.3.2 Back up to within a set tolerance of the wall, maintaining eastern orientation.
```
Comment (AdamuKaapan) [inline Top-Level Requirements.md:85]: 9.3.3 Move the digging arm out of the bin to the predefined dumping position.

Comment (AdamuKaapan) [reply Top-Level Requirements.md:85]: (this is my proposed addition here, feel free to change it to a way that's better phrased)

File: Top-Level Requirements.md [hunk 8]
```diff
@@ -1,63 +1,86 @@
...
+    * 9.4.3 Raise the mining bin and dump the regolith at a set speed.
```
Comment (AdamuKaapan) [inline Top-Level Requirements.md]: This should be 9.3.3

**Review by AdamuKaapan — CHANGES_REQUESTED** [review]
Added a couple extra requirements to solve the bin running into the digging arm during the dumping procedure. Also added a comment regarding a small misnumbering issue on 9.3.3 (which is currently written as 9.4.3).

File: Top-Level Requirements.md (hunk 6 above)
Comment (AdamuKaapan) [reply Top-Level Requirements.md]: Upon further investigation of the system architecture, there are two ways to do this: the first is as I proposed initially, where the request to move the arm goes through the digging action server. The other option is that we can actually remove this extra requirement and have the dumping system interact with the Arm Action Server directly to move the arm, removing the middleman.

The only argument against this alternative option is that it involves bypassing another system's control of this action, but I'm not too concerned about this as digging and dumping should never be operating at the same time, so I'd propose actually not including this requirement and just letting the dumping system interface with the arm action server directly.

I'll leave the original addition here @RyanBerge just for reference, but please note that my updated proposal is to actually not include it.

**Pull request merged**